### ISSUE TITLE
Add support for AMD AOCL libraries

### ISSUE THREAD

**Issue by dwdoerf** [wavefunction91/GauXC#41]
It would be beneficial to support the AMD AOCL LAPACK/BLAS libraries libblis.so and libflame.so, as opposed to using Intel's MKL for AMD CPUs. This will be especially beneficial for NERSC's Perlmutter, which will use AMD Milan CPUs. 

If you want to develop on Cori GPU's A100 nodes, I have installed AOCL and it is available as a module:
```
$ module use /global/cfs/cdirs/mpccc/dwdoerf/cori-gpu/modulefiles
$ module load aocl
```  
I've tried to simply specify them using the following cmake parameters, but if fails in test LAPACK:
```
cgpu20:GauXC$ cat build-aocl.sh
#!/bin/bash
#
#Currently Loaded Modulefiles:
#  1) esslurm         2) dgx/1.0         3) cmake/3.18.2    4) gcc/8.3.0       5) jdk/1.8.0_202   6) cuda/11.0.2     7) aocl/2.2
#

set -x
BUILD_DIR=install_aocl

#  -DOpenBLAS_INCLUDE_DIR=${AOCL_HOME}/include \
cmake -H. \
  -B${BUILD_DIR} \
  -DGAUXC_ENABLE_MPI=OFF \
  -DGAUXC_ENABLE_CUDA=ON \
  -DCMAKE_CUDA_ARCHITECTURES=80 \
  -DGAUXC_ENABLE_MAGMA=OFF \
  -DReferenceBLAS_LP64_LIBRARIES=${AOCL_HOME}/lib/libblis.so \
  -DReferenceLAPACK_LP64_LIBRARIES=${AOCL_HOME}/lib/libflame.so

cmake --build ./${BUILD_DIR}

cgpu20:GauXC$ ./build-aocl.sh 
+ BUILD_DIR=install_aocl
+ cmake -H. -Binstall_aocl -DGAUXC_ENABLE_MPI=OFF -DGAUXC_ENABLE_CUDA=ON -DCMAKE_CUDA_ARCHITECTURES=80 -DGAUXC_ENABLE_MAGMA=OFF -DReferenceBLAS_LP64_LIBRARIES=/global/project/projectdirs/mpccc/dwdoerf/cori-gpu/aocl-linux-gcc/2.2/lib/libblis.so -DReferenceLAPACK_LP64_LIBRARIES=/global/project/projectdirs/mpccc/dwdoerf/cori-gpu/aocl-linux-gcc/2.2/lib/libflame.so
-- The C compiler identification is GNU 7.4.1
-- The CXX compiler identification is GNU 7.4.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The CUDA compiler identification is NVIDIA 11.0.194
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Check for working CUDA compiler: /usr/common/software/osuse15_dgx/cuda/11.0.2/bin/nvcc - skipped
-- Detecting CUDA compile features
-- Detecting CUDA compile features - done
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- GauXC Disabling MPI
-- Could not find IntegratorXX... Building
-- INTEGRATORXX REPO = https://github.com/wavefunction91/IntegratorXX.git
-- INTEGRATORXX REV  = 02ca3b0884a0261d8125cc2ec8397dc3166f2147
-- Performing Test INTEGRATORXX_HAS_NO_MISSING_BRACES
-- Performing Test INTEGRATORXX_HAS_NO_MISSING_BRACES - Success
-- Could not find ExchCXX... Building
-- EXCHCXX REPO = https://github.com/wavefunction91/ExchCXX.git
-- EXCHCXX REV  = 8966c818fe7531f3e85c0d8bc600bc310a99ba65
-- Setting (unspecified) option CMAKE_BUILD_TYPE: Release
-- Setting (unspecified) option BUILD_SHARED_LIBS: OFF
-- Setting option BUILD_TESTING: OFF
-- Setting (unspecified) option BUILD_FPIC: ON
-- Setting (unspecified) option NAMESPACE_INSTALL_INCLUDEDIR: /
-- Setting (unspecified) option ENABLE_GENERIC: OFF
-- Setting (unspecified) option ENABLE_XHOST: ON
-- Performing Test CMAKE_C_FLAGS [-xHost] - Failed
-- Performing Test CMAKE_C_FLAGS [-march=native] - Failed
CMake Warning at install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:62 (message):
  Option unfulfilled as none of -xHost;-march=native valid
Call Stack (most recent call first):
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:105 (add_C_or_CXX_flags)
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:136 (add_C_flags)
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:154 (add_flags)
  install_aocl/_deps/libxc-src/CMakeLists.txt:25 (option_with_flags)


-- Performing Test CMAKE_CXX_FLAGS [-xHost] - Failed
-- Performing Test CMAKE_CXX_FLAGS [-march=native] - Failed
CMake Warning at install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:62 (message):
  Option unfulfilled as none of -xHost;-march=native valid
Call Stack (most recent call first):
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:114 (add_C_or_CXX_flags)
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:139 (add_CXX_flags)
  install_aocl/_deps/libxc-src/cmake/psi4OptionsTools.cmake:154 (add_flags)
  install_aocl/_deps/libxc-src/CMakeLists.txt:25 (option_with_flags)


-- Setting (unspecified) option ENABLE_FORTRAN: OFF
-- Setting (unspecified) option DISABLE_VXC: OFF
-- Setting (unspecified) option DISABLE_FXC: OFF
-- Setting (unspecified) option DISABLE_KXC: ON
-- Setting (unspecified) option DISABLE_LXC: ON
-- Performing Test standard_math_library_linked_to_automatically
-- Performing Test standard_math_library_linked_to_automatically - Failed
-- Performing Test standard_math_library_linked_to_as_m
-- Performing Test standard_math_library_linked_to_as_m - Success
-- Performing Test HAVE_CBRT
-- Performing Test HAVE_CBRT - Success
-- Version: Full 5.0.0
-- SO Version: Full 9:0:0 Major 9
-- Found Perl: /usr/bin/perl (found version "5.26.1") 
-- Found CUDAToolkit: /usr/common/software/osuse15_dgx/cuda/11.0.2/include (found version "11.0.194") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- BLAS_LIBRARIES Not Given: Will Perform Search
-- Checking if OpenMP is GNU
-- Checking if OpenMP is GNU -- YES
-- Could NOT find IntelMKL (missing: IntelMKL_LIBRARIES IntelMKL_INCLUDE_DIR) 
-- Could NOT find IBMESSL (missing: IBMESSL_LIBRARIES IBMESSL_INCLUDE_DIR) 
-- Could NOT find BLIS (missing: BLIS_INCLUDE_DIR) 
-- Could NOT find OpenBLAS (missing: OpenBLAS_INCLUDE_DIR) 
-- Found ReferenceBLAS: /global/project/projectdirs/mpccc/dwdoerf/cori-gpu/aocl-linux-gcc/2.2/lib/libblis.so   
-- Performing Test BLAS_LOWER_UNDERSCORE
-- Performing Test BLAS_LOWER_UNDERSCORE -- found
-- Found BLAS: TRUE   
-- LAPACK_LIBRARIES Not Given: Checking for LAPACK in BLAS
-- Performing Test LAPACK_LOWER_UNDERSCORE
-- Performing Test LAPACK_LOWER_UNDERSCORE -- not found
-- Performing Test LAPACK_LOWER_NO_UNDERSCORE
-- Performing Test LAPACK_LOWER_NO_UNDERSCORE -- not found
-- Performing Test LAPACK_UPPER_UNDERSCORE
-- Performing Test LAPACK_UPPER_UNDERSCORE -- not found
-- Performing Test LAPACK_UPPER_NO_UNDERSCORE
-- Performing Test LAPACK_UPPER_NO_UNDERSCORE -- not found
-- BLAS Does Not Have A Full LAPACK Linker -- Performing Search
-- Found ReferenceLAPACK: /global/project/projectdirs/mpccc/dwdoerf/cori-gpu/aocl-linux-gcc/2.2/lib/libflame.so  found components: lp64 
-- Performing Test LAPACK_LOWER_UNDERSCORE
-- Performing Test LAPACK_LOWER_UNDERSCORE -- not found
-- Performing Test LAPACK_LOWER_NO_UNDERSCORE
-- Performing Test LAPACK_LOWER_NO_UNDERSCORE -- not found
-- Performing Test LAPACK_UPPER_UNDERSCORE
-- Performing Test LAPACK_UPPER_UNDERSCORE -- not found
-- Performing Test LAPACK_UPPER_NO_UNDERSCORE
-- Performing Test LAPACK_UPPER_NO_UNDERSCORE -- not found
CMake Error at /global/common/sw/cray/cnl7/haswell/cmake/cray-cnl7-haswell/gcc-8.3.0/cmake/3.18.2/iteh6ngn/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find LAPACK (missing: LAPACK_LINK_OK)
Call Stack (most recent call first):
  /global/common/sw/cray/cnl7/haswell/cmake/cray-cnl7-haswell/gcc-8.3.0/cmake/3.18.2/iteh6ngn/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  install_aocl/_deps/linalg-cmake-modules-src/FindLAPACK.cmake:153 (find_package_handle_standard_args)
  src/integrator/host/gauxc-host_integrator.cmake:1 (find_package)
  src/integrator/CMakeLists.txt:10 (include)


-- Configuring incomplete, errors occurred!
See also "/global/homes/d/dwdoerf/src/cori-dgx/GauXC/install_aocl/CMakeFiles/CMakeOutput.log".
See also "/global/homes/d/dwdoerf/src/cori-dgx/GauXC/install_aocl/CMakeFiles/CMakeError.log".
+ cmake --build ./install_aocl
gmake: *** No targets specified and no makefile found.  Stop.
```

**Comment by wavefunction91** [comment]
@dwdoerf There's nothing in GauXC that precludes the usage of BLAS/FLAME, its whether or not the CMake discovery module will pick up on it. BLIS will already work (its a part of our CI), I just haven't added explicit support for FLAME. I'll leave this open for now, but I'll also open an issue on the repo that this actually has to get fixed in
https://github.com/wavefunction91/linalg-cmake-modules/issues/6

**Comment by wavefunction91** [comment]
Closed via #47
